cfme/cloud/flavor.py

```diff
@@ -97,25 +98,106 @@ def is_displayed(self):
+    category = BootstrapSelect("tag_cat")
+    tag = BootstrapSelect("tag_add")
     save = Button('Save')
     reset = Button('Reset')
     cancel = Button('Cancel')
 
 
+class FlavorCollection(Navigatable):
+    def instantiate(self, name, provider):
+        return Flavor(
+            name,
+            provider,
+            collection=self
+        )
+
+    def add_tag(self, tag, *flavors):
+        """Add given tag to flavor collections
+
+        Args:
+            tag: tuple describing tags
+            flavors: list of :py:class:`cfme.cloud.flavor.Flavor` objects
+        """
+        flavors = list(flavors)
+        checked_flavors = []
+        view = navigate_to(FlavorCollection, 'All')
+        # double check we're in List View
+        view.toolbar.view_selector.select('List View')
+        if not view.entities.table.is_displayed:
+            raise ValueError('No flavor found')
+        for row in view.entities.table:
+            for flavor in flavors:
+                if flavor == row.name.text:
+                    checked_flavors.append(flavor)
+                    row[0].check()
+                    break
+            if set(flavors) == set(checked_flavors):
+                break
+        if set(flavors) != set(checked_flavors):
+            raise ValueError('Some flavors were not found in the UI')
+        view.toolbar.policy.item_select('Edit Tags')
+        mixins.add_tag(tag, navigate=False)
+
+    def remove_tag(self, tag, *flavors):
+        """Remove given tag from flavor collections
+
+        Args:
+            tag: tuple describing tags
+            flavors: list of :py:class:`cfme.cloud.flavor.FlavorCollection` objects
+        """
+        flavors = list(flavors)
+        checked_flavors = []
+        view = navigate_to(FlavorCollection, 'All')
+        # double check we're in List View
+        view.toolbar.view_selector.select('List View')
+        if not view.entities.table.is_displayed:
+            raise ValueError('No flavor found')
+        for row in view.entities.table:
+            for flavor in flavors:
+                if flavor == row.name.text:
+                    checked_flavors.append(flavor)
+                    row[0].check()
+                    break
+            if set(flavors) == set(checked_flavors):
+                break
+        if set(flavors) != set(checked_flavors):
+            raise ValueError('Some flavors were not found in the UI')
+        view.toolbar.policy.item_select('Edit Tags')
+        mixins.remove_tag(tag)
+
+
 class Flavor(Navigatable):
-    """
-    Flavor class to support navigation
-    """
-    _param_name = "Flavor"
+    """Flavor class to support navigation"""
 
-    def __init__(self, name, provider, appliance=None):
+    def __init__(self, name, provider, collection=None):
+        """Base class for flavor"""
         self.name = name
         self.provider = provider
-        Navigatable.__init__(self, appliance=appliance)
```

Comment:
Why remove the Navigatable init?  You need to keep it for now and pass the collection's appliance.